eZ/Publish/Core/Persistence/Cache/LocationHandler.php

```diff
@@ -186,12 +186,21 @@ public function unHide($locationId)
     public function swap($locationId1, $locationId2)
     {
         $this->logger->logCall(__METHOD__, array('location1' => $locationId1, 'location2' => $locationId2));
-        $return = $this->persistenceHandler->locationHandler()->swap($locationId1, $locationId2);
+        $locationHandler = $this->persistenceHandler->locationHandler();
+
+        $return = $locationHandler->swap($locationId1, $locationId2);
+
+        $location1 = $locationHandler->load($locationId1);
```

Comment:
The latter seems a little bit more opportunistic to me, updated in https://github.com/ezsystems/ezpublish-kernel/pull/1781/commits/d73f85137751f9120b84c69d948e13a5d25c28fb